' microhomology_score.R

Computes HRDetect microhomology scores as described by Davies et al. (2017).
https://www.ncbi.nlm.nih.gov/pubmed/28288110.

Required input is annotated indel calls in TSV format from flank_indels.R.

Usage: microhomology_score.R -i INDEL -o OUTPUT

Options:
    -i --indels INDEL       Annotated indel table, as generated by flank_indels.R
    -o --output OUTPUT      Path to output table of microhomology scores

Example:
    Rscript microhomology/flank_indels.R -v indel_calls.vcf -o annotated_indels.tsv
    Rscript microhomology/compute_microhomology_score.R -i annotated_indels.tsv -o microhomology_results.tsv
' -> doc

library(docopt)
args <- docopt(doc)

library(tidyverse)

microhomology <- read_tsv(args[['indels']], col_types=cols(is_microhomology=col_logical())) %>%
    summarise(
        deletion_microhomology_count = sum(is_microhomology),
        deletion_count = n()
    ) %>%
    mutate(
        deletion_microhomology_proportion = if_else(deletion_count == 0, 0, deletion_microhomology_count / deletion_count)
    )

write_tsv(microhomology, args[['output']])
